scrapy/core/downloader/webclient.py

```diff
@@ -7,21 +7,31 @@
 
 from scrapy.http import Headers
 from scrapy.utils.httpobj import urlparse_cached
+from scrapy.utils.python import to_bytes
 from scrapy.responsetypes import responsetypes
 
 
 def _parsed_url_args(parsed):
+    # Assume parsed is urlparse-d from Request.url,
+    # which was passed via safe_url_string and is ascii-only.
```

Comment:
See also: https://github.com/scrapy/scrapy/issues/833. I think that's fine to leave it as-is for now, but we should keep this use case in mind.